### DISCUSSION TITLE
小程序基础库版本依赖

### DISCUSSION THREAD

**Discussion by lixingdecai** [Tencent/tdesign-miniprogram#192] [General]
### 目前TD 小程序设置

TD小程序组件库作为基础通用组件，为了覆盖更多的用户，满足更多的业务场景，我们会尽量兼容更多的基础库版本。

TD小程序组件库组件库设置最低基础库版本为 2.6.3，微信大盘灰度用户占比 99.7%（截止2022.1）。各个项目小程序使用受众不同，占比也会有所差异，具体可以查看mp管理后台。

**建议**：使用项目基础库版本不低于2.6.3

#### 影响：
1.  如果用户微信版本较老，导致其基础库小于设置的最低基础库，会提示用户更新微信app，无法访问小程序内容。
2. 如果业务项目设置基础库低于组件库基础库版本，会导致一些特性在少部分用户上异常无法使用。

### 变更历史

2021.12，基础库2.6.3，经问题反馈和微信官方回复确认，observers存在兼容性问题，建议设置为2.6.3
2021.10，基础库2.6.1，使用observers 特性，原observer 监听性能api的性能问题较大。

### 参考文档

- 基础库版本分布：https://developers.weixin.qq.com/miniprogram/dev/framework/client-lib/version.html
- 基础库版本特性汇总（欢迎大家一起维护）：https://doc.weixin.qq.com/sheet/e3_m_ORDfxiTWKzav?scode=AJEAIQdfAAoqCkEASJAOMAZwatACg&tab=xv1ul2

ps: 欢迎大家反馈基础库版本的问题，一起将问题反馈给更多同学🌹

**Comment by anlyyao** [comment]
【Input】【Button】【Switch】【Checkbox】【Radio】等组件与Form表单联用时， 若想使自定义组件有类似于表单控件的行为，需使用内置 behaviors，对于 form 组件，目前可以自动识别：wx://form-field、wx://form-field-group、wx://form-field-button。其中wx://form-field-group从基础库版本2.10.2开始提供支持，wx://form-field-button从基础库版本 2.10.3开始提供支持，综上考虑建议将基础库设置为2.10.4.
参考：[https://developers.weixin.qq.com/miniprogram/dev/framework/custom-component/behaviors.html]()
